cli/azd/pkg/commands/pipeline/pipeline.go

```diff
@@ -84,75 +96,123 @@ func folderExists(folderPath string) bool {
+		lastUsedProvider, configExists := env.Values[envPersistedKey]
+		if configExists {
+			// Setting override value based on last run. This will force detector to use the same
+			// configuration.
+			overrideWith = lastUsedProvider
+		}
+		// Figure out what is the expected provider to use for provisioning
+		prj, err := project.LoadProjectConfig(azdContext.ProjectPath(), env)
+		if err != nil {
+			return nil, nil, fmt.Errorf("finding pipeline provider: %w", err)
+		}
+		if prj.Pipeline.Provider != "" {
+			overrideWith = prj.Pipeline.Provider
+		}
 
-	if hasAzDevOpsFolder && !hasGitHubFolder {
-		// Azdo only
-		return &AzdoHubScmProvider{}, &AzdoCiProvider{}, nil
 	}
 
-	// Both folders exist. Prompt to select SCM first
-	scmElection, err := console.Select(ctx, input.ConsoleOptions{
-		Message: "Select what SCM provider to use",
-		Options: []string{
-			"GitHub",
-			"Azure DevOps",
-		},
-		DefaultValue: "GitHub",
-	})
-
-	if err != nil {
-		return nil, nil, err
+	// Check override errors for missing folder
+	if overrideWith == gitHubLabel && !hasGitHubFolder {
+		return nil, nil, fmt.Errorf("%s folder is missing. Can't use selected provider.", githubFolder)
+	}
+	if overrideWith == azdoLabel && !hasAzDevOpsFolder {
+		return nil, nil, fmt.Errorf("%s folder is missing. Can't use selected provider.", azdoFolder)
+	}
+	// using wrong override value
+	if overrideWith != "" && overrideWith != azdoLabel && overrideWith != gitHubLabel {
+		return nil, nil, fmt.Errorf("%s is not a known pipeline provider.", overrideWith)
 	}
 
-	if scmElection == 1 {
-		// using azdo for scm would only support using azdo pipelines
-		return &AzdoHubScmProvider{}, &AzdoCiProvider{}, nil
+	// At this point, we know that override value has either:
+	// - github or azdo value
+	// - OR is not set
+	// And we know that github and azdo folders are present.
+	// checking positive cases for overriding
+	if overrideWith == azdoLabel || hasAzDevOpsFolder && !hasGitHubFolder {
+		// Azdo only either by override or by finding only that folder
+		_ = savePipelineProviderToEnv(azdoLabel, env)
+		return createAzdoScmProvider(env, azdContext), createAzdoCiProvider(env, azdContext), nil
 	}
 
-	// GitHub selected for SCM, prompt for CI provider
-	ciElection, err := console.Select(ctx, input.ConsoleOptions{
-		Message: "Select what CI provider to use",
-		Options: []string{
-			"GitHub Actions",
-			"Azure DevOps Pipelines",
-		},
-		DefaultValue: "GitHub Actions",
-	})
+	// Both folders exists and no override value. Default to GitHub
+	// Or override value is github and the folder is available
+	_ = savePipelineProviderToEnv(gitHubLabel, env)
+	return &GitHubScmProvider{}, &GitHubCiProvider{}, nil
+}
 
+func savePipelineProviderToEnv(provider string, env *environment.Environment) error {
+	env.Values[envPersistedKey] = provider
+	err := env.Save()
 	if err != nil {
-		return nil, nil, err
+		return err
 	}
+	return nil
+}
```

Comment:
There is a similar new method in the `azdo/utils` package.

Reply:
The one in azdo/utils was is a private fun used only in the azdo implementrtion. We could move it to the environment package to make generic. This method in the pipeline command is there to avoid needing to bring in the azdo package, since the command is provide agnostic (other than selecting a provider) and leaves the underlying implementation to the provider itself.